static/src/components/setting/index.vue

```diff
@@ -0,0 +1,365 @@
+        content = this.externalConfigForm[key];
+        inputFields.push(content);
+        content.fields.forEach(function (f) {
+          configFields[f.name] = null;
+        });
+      }
+      this.externalConfigDict = configFields;
+      this.updateExternalConfigDict();
+      return inputFields;
+    },
+
+    updateExternalConfigDict () {
+      for (let key of Object.keys(this.externalConfig)) {
+        if (typeof this.externalConfig[key] === 'object') {
+          Object.assign(this.externalConfigDict, this.externalConfig[key]);
+          } else this.externalConfigDict[key] = this.externalConfig[key];
+        }
+    },
+
+    constructExternalConfigFromDict () {
+      let config = {};
+      let dict = this.externalConfigDict;
+      for (let [key, content] of Object.entries(this.externalConfigForm)) {
+        let _cf = {};
+        content.fields.forEach(function (f) {
+          _cf[f.name] = dict[f.name];
+        });
+        config[key] = _cf;
+      }
+      return config;
+    },
+
+    add (event, ur) {
+      Vue.set(this.assignee, ur.id, ur);
+      this.assignee[ur.id] = ur;
+    },
+
+    remove (event, id) {
+      Vue.delete(this.assignee, id);
+    },
+
+    filter () {
+      if (!this.search.length) {
+        this.searchResult = this.allUsers;
+      } else {
+        this.searchResult = this.allUsers.filter(
+          user => user.fullname.toLowerCase().includes(this.search.toLowerCase()));
+      }
+    },
+
+    async save () {
+      let access = [];
+       for (let [key, value] of Object.entries(this.accessLevels)) {
+         if (value) {
+           access.push(key);
+         }
+       }
+      let assigneeId = Object.keys(this.assignee);
+      try {
+        const res = await fetch(window.location.pathname, {
+          method: 'POST',
+          headers: {
+            'content-type': 'application/json',
+            'X-CSRFToken': this.csrfToken
+          },
+          credentials: 'same-origin',
+          body: JSON.stringify({ 'project': {
+            config: this.constructExternalConfigFromDict(),
+            data_access: access,
+            select_users: assigneeId
+          } })
+        });
+        if (res.ok) {
+          window.pybossaNotify('Project data Saved.', true, 'success');
+        } else {
+          window.pybossaNotify('An error occurred.', true, 'error');
+        }
+      } catch (error) {
+        window.pybossaNotify('An error occurred.', true, 'error');
+      }
+    }
+  }
+};
+
+</script>
+<style>
+
+.scroll #users:hover {background-color: #ddd}
+.scroll #users {padding-left: 10px; box-sizing: border-box;}
+.scroll p {
+  color: black;
+  padding: 2px 2px;
+  text-decoration: none;
+  display: block;
+}
+.scroll {
+  overflow-x: hidden;
+  max-height: 150px;
+  overflow-y: scroll;
+}
+.form-check-label p {
+  margin-right:10px
+}
+.title {
+  border-top: 1px ridge silver
+}
+#assign-user-column {
+  border-right: 1px ridge silver; width:50%
```

Comment:
in general there seems to be a lot of specific hardcoded style (colors, line widths, borders, margin, paddings). In my opinion this will make it harder to keep the style consistent, especially if at some point the global style is changed. Again my opinion it's better instead use scss, reuse already defined colors and styles if possible and create appropriate classes when needed.

Reply:
agree, already fixed in latest commit.